### PR TITLE
No HI

### PR DESCRIPTION
**Pull request by Pmaene**
Added support to exclude Hearing Impaired subtitles from the results.

### CODE REVIEW

File: lib/addic7ed/subtitle.rb [hunk 1]
```diff
@@ -18,8 +19,13 @@ def to_s
       "#{url}\t->\t#{version} (#{language}, #{status}) [#{downloads} downloads]#{" (via #{via})" if via}"
     end
 
-    def works_for?(version = '')
-      is_completed? and is_compatible_with? version
+    def works_for?(version = '', no_hi = false)
+      hi_works = true
+      if @hi and no_hi
+        hi_works = false
+      end
```
Comment (michaelbaudino) [inline lib/addic7ed/subtitle.rb]: This whole logic could simply be replaced by `hi_works = !no_hi || !@hi`

File: lib/addic7ed/episode.rb [hunk 2]
```diff
@@ -48,10 +48,10 @@ def find_subtitles(lang)
       @subtitles[lang] = parser.extract_subtitles
     end
 
-    def find_best_subtitle(lang)
+    def find_best_subtitle(lang, no_hi)
```
Comment (michaelbaudino) [inline lib/addic7ed/episode.rb]: Can you make this argument optional like in `best_subtitle` ?

File: lib/addic7ed/episode.rb [hunk 3]
```diff
@@ -23,18 +23,18 @@ def subtitles(lang = 'fr')
...
-    def best_subtitle(lang = 'fr')
+    def best_subtitle(lang = 'fr', no_hi = false)
       check_language_availability(lang)
-      find_best_subtitle(lang) unless @best_subtitle and @best_subtitle[lang]
+      find_best_subtitle(lang, no_hi) unless @best_subtitle and @best_subtitle[lang]
       return @best_subtitle[lang]
     end
 
-    def download_best_subtitle!(lang, http_redirect_limit = 8)
+    def download_best_subtitle!(lang, no_hi, http_redirect_limit = 8)
```
Comment (michaelbaudino) [inline lib/addic7ed/episode.rb]: Having `no_hi` parameter optional here would be great for initial call.

**Comment by michaelbaudino** [comment]
@Pmaene: same comment as on #12 : I'm absolutely confused I missed this PR months ago, but it's again very good work :+1: 
If you can address my comments, I'll be very happy to merge and have this new feature :shipit:

**Comment by michaelbaudino** [comment]
Ho, also, it does not merge :disappointed:

**Comment by michaelbaudino** [comment]
And would it be a lot of work to add a `--hi-only` option for a11y ?

**Comment by Pmaene** [comment]
Here you go 😊. I noticed that my test case was not correct, so I have also updated that one. I added "filename_no_hi", although it would work with the original "filename" with the changes from #12. I chose English instead of French, because English has both kinds of subs (HI and no HI).

I think it would be quite a bit of work to add an --hi-only, as we'd need to refactor everything to support either no preference, only HI or only not HI.

File: spec/lib/addic7ed/episode_spec.rb [hunk 4]
```diff
@@ -108,34 +116,34 @@
 
     it 'gets the best subtitle candidate with a network request' do
       expect(episode).to receive(:best_subtitle).once.and_call_original
-      episode.download_best_subtitle!('fr')
+      episode.download_best_subtitle!('fr', false)
```
Comment (michaelbaudino) [inline spec/lib/addic7ed/episode_spec.rb]: The second argument here (`false`) can now be removed :smiley: 
Same thing at lines 126, 133, 139, 146, 156 and 162.

**Comment by michaelbaudino** [comment]
Thanks, we're almost there :+1: :smile:
Can you merge `master` branch into this one? (because it has conflicts right now and thus I cannot merge it)

**Comment by Pmaene** [comment]
Removed the argument and merged all upstream changes 😃.

**Comment by michaelbaudino** [comment]
Thanks a lot :+1:

**Pull request merged**